doc/whats_new/upcoming_changes/sklearn.metrics/31041.fix.rst

```diff
@@ -0,0 +1,3 @@
+- In :mod:`metrics._regession` updated set_huber_delta raise an value error when all weights are zero.
+- In :mod:`metrics._regession` updated median_absolute_error raise an value error when all weights are zero.
+- In :mod:`metrics._regession` updated d2_pinball_score raise an value error when all weights are zero.
```

Comment:
I think these can be a single entry that lists all 3 functions.

Please use the syntax: 
```
:func:`~metrics._regession.updated d2_pinball_score` 
```
etc

Reply:
Please be explicit and use `sample_weight`, e.g.,
```suggestion
- Updated :func:`~metrics._regession.set_huber_delta` to raise a value error when
  `sample_weight` are all zeros.
```